More on re-brand fixes and improvements #508

th1nkNXT opened the pull request:
Account wasn't shown anymore in left sidebar of the classic wallet. Restored old code. Fixed.

**Important:**
1. There is still the wording BURST in copy_send_burst_link (locales). If you want i can fix that, too.
2. As i do not know the whole existing functionality i am not sure if the functionality for `copy_account_id` is correct. It uses the value of `account_rs` as i thought should lead to a copy of the RS address like S-59P9-XXXX-XXXX-XXXXX but if i click on the link to copy the data i get the RS address followed by another string like S-59P9-XXXX-XXXX-XXXXX-MANYCHARSANDNUMBERS

![image](https://user-images.githubusercontent.com/18901038/122825655-84f4a700-d2e2-11eb-8a78-8d1553593e7a.png)

Question about reference information in the commit: I used pull request #508 as a reference. I am not sure if this was correct. Normally i reference issue numbers. If there is something to improve here just give me feedback so i will take care of that in the future. Thanks.

jjos2372 commented:
The long address you mentioned is the extended address https://github.com/burst-apps-team/CIPs/blob/master/cip-0026.md to be used when the account is not *activated* yet.

We were trying to move away from the "menu" solution, making the user to click to see the options. The problem is in doing that we let the address out. If you have time, please try to make all the options always visible but also with the address.
PS. The classic wallet is in *maintenance* mode only.

th1nkNXT commented:
Thanks for the explanation. Ok, i will look into it.

th1nkNXT commented:
@jjos2372 
I have finished my work. It is just a proposal. What i did so far:

1. Removed sidebar dropdown
2. Menu items are always visible
3. Changed styling to default nav-pills nav-stacked bootstrap navbar
4. New: Moved copy operation for RS address to the shown address itself (This fixes also an issue with missing i18n text constants for that case. Previously the wrong constant `copy_account_id` was used and the `copy_account_rs` constant exists in English only). I used the term "Copy" as a tooltip here now as simple solution (no i18n).
5. New: Increased font size of RS address.
6. Removed obsolete css stylings.

Look & feel is not max. optimized but i think it is okay:
![image](https://user-images.githubusercontent.com/18901038/124181739-34393700-dab6-11eb-92dc-dcfba64e653f.png)
![image](https://user-images.githubusercontent.com/18901038/124181781-41562600-dab6-11eb-8030-9e43b467c2f0.png)
![image](https://user-images.githubusercontent.com/18901038/124181819-4dda7e80-dab6-11eb-9338-4682b97e85dc.png)

Tested with Firefox and Chrome (Desktop, Windows).
Please give me some feedback on this. 
I do not know if i should mention you via your username or not. If you get too much messages just say it ;-) I wasn't sure if you see my comments here.

th1nkNXT commented:
One more thing (which should be handled maybe with a separate task):
The texts are not initialized correctly during the first load at least if the language is English (default).
The i18n routine is not getting called in that case:
![image](https://user-images.githubusercontent.com/18901038/124329251-708a9700-db8b-11eb-8ebe-35ce0a810774.png)

To have a quick fix here i renamed the default text:
![image](https://user-images.githubusercontent.com/18901038/124329359-944ddd00-db8b-11eb-8599-716bbbc33736.png)
see my last commit for this.

jjos2372 commented:
Great, thanks. I want to merge this on the new 'develop' branch. Let me know if I should open another PR for that or if you want to do it.
PS. Is the 'extended' address copied on your fix when the account is still 'unknown' by the network? That was the plan, that the copy will get you the RS address if the account is known, or the extended one if not. Let me know.

th1nkNXT commented:
1. i18n initialization problem when language is English (default) => I fixed it already by modifying the default text, so there shouldn't be a problem for the end user right now. The technical thing about the i18n initialization should be fixed with a new issue / PR as it is not related to this task / problem. But as you said The Classic Wallet is in maintenance mode only so i think priority will be low. So maybe we keep it like it is as long as no one has a real problem here?
2. `account_rs` is used (which uses `BRS.accountRSExtended` if there is an error code in `BRS.accountInfo.errorCode`). Is it correct, that an account is in error state as long as the node isn't fully synced yet or how long is an account unknown to the network? Sorry for asking dump questions ;-) In my case my test account (my node isn't fully synced yet) shows the following data when using the copy operation:
![image](https://user-images.githubusercontent.com/18901038/124503414-c8acdd80-ddc5-11eb-9a89-a26f282760b2.png)

RS with public key:
`S-59P9-R3TA-Y9RF-6D5C8-1W0LZK3OGSCM33M1IFD90UTM41XTS8V8MM662XA20Z95N5J0C0`

JS code used for extended address:
`BRS.accountRSExtended = BRS.accountRS + '-' + new BigNumber(BRS.publicKey, 16).toString(36).toUpperCase();`

This seems to be correct for me (i am not sure if the numerical representation is correct but that wasn't touched by me). If there is something more i could test please let me know.

jjos2372 commented:
Thank you, merged on the develop branch, so closing here.
